mlflow/tracking/client.py

```diff
@@ -142,6 +142,57 @@ def get_metric_history(self, run_id, key):
         :param key: Metric name within the run
 
         :return: A list of :py:class:`mlflow.entities.Metric` entities if logged, else empty list
+
+        .. code-block:: python
+            :caption: Example
+
+            from mlflow.tracking import MlflowClient
+
+            def print_metric_info(history):
+                for m in history:
+                    print("name: {}".format(m.key))
+                    print("value: {}".format(m.value))
+                    print("timestamp: {}".format(m.timestamp))
+                    print("--")
+
+            # Create a run under the default experiment (whose id is "0"). Since this is low-level
+            # CRUD operation, the method will create a run. To end the run, you'll have
+            # to explicitly end it.
+            client = MlflowClient()
+            experiment_id = "0"
+            run = client.create_run(experiment_id)
+            print("run_id: {}".format(run.info.run_id))
+            print("--")
+
+            # Log couple of metrics, update their initial value, and fetch each
+            # logged metrics' history.
+            for k, v in [("m1", 1.5), ("m2", 2.5)]:
+                client.log_metric(run.info.run_id, k, v)
+                client.log_metric(run.info.run_id, k, v + 1)
```

Comment:
```suggestion
                client.log_metric(run.info.run_id, k, v, step=0)
                client.log_metric(run.info.run_id, k, v + 1, step=1)
```

because some users might think "why overwrite the same metric here"?

Reply:
Do we need to specify step=0 since it defaults to 0 if the step is omitted in the argument? So these two statements should achieve the same result: 

```
client.log_metric(run.info.run_id, k, v)
client.log_metric(run.info.run_id, k, v + 1, step=1)
```
Is there a specific reason that I'm missing why you have `step=0`?

Reply:
sorry it's just my preference...

Reply:
> Maybe we should explain a bit more what step=<value> does in the parameter description. What do you think?

I agree, we could do that as a separate (or follow-up) PR.

Reply:
Cool, will fix it, and yes we can make that as a separate PR. There quite a few places where the parameters' semantics are missing, and a beginner may miss its merits.

Reply:
Thanks!